_rules/autocomplete-valid-value-73f2c2.md

```diff
@@ -24,7 +24,7 @@ acknowledgments:
 This rule applies to any HTML `input`, `select` and `textarea` element with an `autocomplete` [attribute value][] that is neither empty (`""`) nor only [ASCII whitespace][], except if one of the following is true:
 
 - **hidden**: the element is not [visible][], and not [included in the accessibility tree][]; or
-- **disabled**: the element has an `aria-disabled` [attribute value][] of `true`; or
+- **disabled**: the element has the `disabled` attribute or the `aria-disabled` [attribute value][] of `true`; or
```

Comment:
This should probably use https://act-rules.github.io/glossary/#disabled-element instead.

Also, this question the need for the last assumption. I essentially understood the rule (+assumption) as looking only at `aria-disabled` because it assumed that if the element is disabled (common meaning), this is reflected in the `aria-disabled` attribute, notably, if `disabled` is set, then `aria-disabled` is also set (by default) (see https://www.w3.org/TR/html-aam-1.0/#html-attribute-state-and-property-mappings / disabled, and actually if trying to set both, then `disabled` seems to take precedence).

I think that the definition (and its assumption) is capturing this idea even better (it was written after this rule) and should be used here also.

Reply:
The disabled element definition, as far as I understand it, applies to elements such as `<button disabled>Submit</button>` or to elements that are in the shadow DOM. It does not seem to apply to elements such as `<button aria-disabled>Submit</button>`. I'm trying to cover both scenarios but I agree that we should use the definition of the disabled element instead of "disabled" attribute. 

Am I right in saying that you're referring to `<button disabled aria-disabled>Submit</button>`?

Reply:
Sorry, I think that the penny dropped, we should use the disabled element definition because `disabled` is mapped to platform-specific accessibility API's as `aria-disabled="true"`, is that correct?

Reply:
The second point in the definition is about `aria-disabled` (and should read "inclusive ancestor", rather than "ancestor" 🙈 )

"shadow including" ancestor means that it indeed apply to elements in the shadow DOM, but not only. `<div><span></span></div>`, the `div` is both an ancestor and a shadow including ancestor (i.e., it's shadow including, not light excluding).

Copying DOM standard, emphasise mine:
> An object A is a shadow-including descendant of an object B, if **A is a descendant of B**, or A’s root is a shadow root and A’s root’s host is a shadow-including inclusive descendant of B.
>
> A shadow-including inclusive descendant is an object or one of its shadow-including descendants.
>
> An object A is a shadow-including ancestor of an object B, if and only if B is a shadow-including descendant of A.
>
> A shadow-including inclusive ancestor is an object or one of its shadow-including ancestors.

So, regular (inclusive) ancestors/descendants are also shadow-including ones.

Reply:
Took a while, but got it in the end. Should be fixed now. Thanks!